### ISSUE TITLE
[Feature]: Stats Leaderboard -- Add placeholder text under wins and losses if you have no wins/losses

### ISSUE THREAD

**Issue by itsalaidbacklife** [cuttle-cards/cuttle#229]
### Feature Summary

When a player is undefeated or has no wins in a given week (or total) their menu for that week will have a blank section under wins/losses. We should add the text 'None' to make this space more naturally

### Detailed Description

![image](https://user-images.githubusercontent.com/6520704/203680466-899fb1fb-0902-4a7e-8f84-e80abe827d76.png)

Instead of a blank section under 'Losses' this could say 'None' so that the section still has text

**Comment by colindaviddick** [comment]
Hey, I'd like to take this on

**Comment by itsalaidbacklife** [comment]
@colindaviddick great, thank you! Please let me know if you have any questions or issues setting up. You can ask questions here in the issue thread for async feedback or hit us up on [discord](https://discord.gg/6vQCFa6nvP) if you'd like to chat with me or other contributors in real-time

**Comment by colindaviddick** [comment]
Thanks, I'm having problems getting Cypress installed. I've installed nvm-windows to get Node 16 installed, but it looks like it's conflicting with another node version manager I'd installed previously and forgot about, so I'm working on sorting that out at the moment.

**Comment by itsalaidbacklife** [comment]
@colindaviddick ah my sympathies, no one likes version wrangling. Are you on windows 10 or 11? We've had folks have difficulty using windows 11 so if that's your OS then I recommend using windows subsystem linux for your setup if possible (should come with your OS)

**Comment by colindaviddick** [comment]
I'm on Windows 10, it's ok I think I'm getting there. 

I haven't tested it out yet as I haven't had the app working, but I was thinking that changing the "playersBeaten" and "playersLostTo" default text to "None" should be all that's needed for this feature?

**Comment by itsalaidbacklife** [comment]
Right general idea but the `playersBeaten` and `playersLostTo` props will always be present and passed from the `StatsLeaderboard.vue` to each `StatsLeaderboardCell.vue` so I think rather than giving the prop a default value we should define one computed prop for each that returns 'None' if the corresponding prop is the empty string and otherwise returns the prop value, then display those computed props in the template rather than the props themselves

**Comment by colindaviddick** [comment]
In the process of fixing Node I've removed npx from my system, so I need to add that or yarn now to get Cypress to work, however, I have the game working, I played a few games and I like it. 

I had no idea this would take me so long, it's 2:30am now and it's my birthday tomorrow so I'm not sure if I'll be able to get anything done tomorrow, if you want to pass it off to someone else I don't mind, otherwise it'll be the weekend before I get a chance to work on this again.

**Comment by itsalaidbacklife** [comment]
@colindaviddick Happy birthday! Sorry setting up node has been such a pain. Hopefully now you'll be able to manage version switching more easily. 

There is no rush, so if you'd still like to complete this issue you are more than welcome to it.

**Comment by colindaviddick** [comment]
I think I've solved the issue but I'm not sure how to actually view the window in the image. I've played the game (pretty cool btw, I like the styling and it's a fun game.) But I'm assuming I need to play a ranked game before I can view the section above in the results? When I do try a ranked game the game just doesn't end, it plays on for a couple more turns and then both players are told that it's not their turn. I'm assuming because it's not getting an expected response from a database?

I think I should maybe bow out of this one now, as to be honest I haven't used Vue.js yet, and I'm having difficulty with Cypress as well, it's failing a lot of tests before I even change anything in the code, presumably because it's not set up correctly. I thought this was going to be far more straightforward than it has been. I'm happy to submit the code that I've created, as I think it should work, but unfortunately have no way at the moment of proving that.

Either way thanks for letting me have a go at it, but I feel like someone else could do this in a lot less time, and I'd prefer to work on things that are more closely related to what I've been learning, and I feel like I'm trying to get my system set up to work on a project that I'm unlikely to spend a lot more time on, since it's built in Vue and I've currently been learning React. Apologies for this, but when I looked at the code before commenting I did think it was going to be very simple.

**Comment by itsalaidbacklife** [comment]
@colindaviddick understood; no worries. Thanks for giving it a shot and for your feedback. I’m sorry it’s been such a hassle setting up—can you help me understand what some of the issues you faced have been?

For reference, the ideal way to test this feature would be with a new test case in the stats page test, which lets you seed data directly so you have ranked match data to play with. 

Regardless thanks for efforts and for helping us find ways to improve the developer experience. Best of luck with your learning!

**Comment by colindaviddick** [comment]
No problem, I don't know how much of this will translate to other people, but I had problems getting the correct version of Node. I had already used nodist for version management, but wasn't really aware of that at the time, so after installing nvm-windows I was causing compatibility problems, so that took a while to diagnose and fix. (I'm also slightly concerned that I've affected compatibility with some of my own projects now)

Cypress was new to me, and some of the tests worked ok but some were timing out or throwing errors. I suspect that wasn't anything to do with the project, and probably something wrong with compatibility with my setup. I haven't come across Cypress before and (not to sound rude) but at the moment I don't want to learn this specifically for this one task. Since it's Vue.js and not a framework I'm more familiar with, I don't see any point for me personally to learn how to use a technology I haven't come across before just for this small task. If/when I start learning Vue I'll definitely come back to Cuttle as I do like the game.

Mainly, the problem is though that I couldn't get to the page I was trying to change, to see if what I had changed was working. Learning Cypress would solve that I'm sure, but just running the stats.spec.js file, while it does show some results now, is still giving a lot of errors as well. I feel like the time and effort spent in getting all of these things sorted is unnecessary if the problems are only on my end, since it's not a big job and for someone who's already working with these technologies could be done a lot quicker and with far less headaches I'm sure.
